flytekit/clients/raw.py

```diff
@@ -96,6 +97,46 @@ def handler(*args, **kwargs):
     return handler
 
 
+def _handle_rpc_error_with_retries(fn):
+    def handler(*args, **kwargs):
+        """
+        Wraps rpc errors as Flyte exceptions and handles authentication the client.
+        Retries most errors. Avoid using for non-idempotent invocations.
+        :param args:
+        :param kwargs:
+        :return:
+        """
+        retries = 3
+        max_wait_time = 1000
+
+        try:
+            for i in range(retries):
+                try:
+                    return fn(*args, **kwargs)
+                except _RpcError as e:
+                    if e.code() == _GrpcStatusCode.UNAUTHENTICATED:
+                        if i == (retries - 1):
+                            # Exit the loop and wrap the authentication error.
+                            raise _user_exceptions.FlyteAuthenticationException(_six.text_type(e))
+                        refresh_handler_fn = _get_refresh_handler(_creds_config.AUTH_MODE.get())
+                        refresh_handler_fn(args[0])
+                    else:
+                        if i == (retries - 1):
+                            raise
+                        else:
+                            # Start with 200ms wait-time and exponentially back-off upto 1 second.
+                            wait_time = min(200 * (2 ** retries), max_wait_time)
+                            _logging.error(f"Non-auth RPC error {e}, sleeping {wait_time}ms and retrying")
+                            time.sleep(wait_time / 1000)
+        except _RpcError as e:
+            if e.code() == _GrpcStatusCode.ALREADY_EXISTS:
```

Comment:
This should not happen for GEt right?

Reply:
Not just for get, hope that this to be used for any mutating calls as well as long as they are idempotent